app/models/profile.rb

```diff
@@ -112,7 +147,7 @@ def name_or_email
   end
 
   def main_topic_or_first_topic
-    main_topic.present? ? main_topic : topic_list.first
+    main_topic.present? ? main_topic : taggings.map { |tagging| tagging.tag.name }.first
```

Comment:
why did you do that?

Reply:
The `topic_list` method created a n+1 query as fas as I remeber.